pkg/common/nats/conn.go

```diff
@@ -74,6 +79,34 @@ func NewNatsConn(ctx context.Context, config commonconfig.EventingNatsConfig) (*
 		opts = append(opts, o)
 	}
 
+	// reconnection options
+	if config.ConnOpts.RetryOnFailedConnect {
```

Comment:
I think you need to check for `config.ConnOpts != nil`

Reply:
fixed